app/scripts/controllers/reports/donationsReport.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
 angular.module('bsis')
-  .controller('DonationsReportCtrl', function($scope, $location, $log, $filter, $routeParams, ICONS, PERMISSIONS, ReportsService) {
+  .controller('DonationsReportCtrl', function($scope, $log, $filter, ReportsService, DATEFORMAT) {
 
-    $scope.icons = ICONS;
-    $scope.permissions = PERMISSIONS;
+    var master = {
+      startDate: moment().subtract(7, 'days').startOf('day').toDate(),
+      endDate: moment().endOf('day').toDate()
+    };
+
+    $scope.dateFormat = DATEFORMAT;
+    $scope.search = angular.copy(master);
+
+    $scope.clearSearch = function(form) {
+      $scope.search = angular.copy(master);
+      form.$setPristine();
+      form.$setUntouched();
+      $scope.gridOptions.data = [];
+    };
+
+    $scope.getReport = function(selectPeriodForm) {
+      if (selectPeriodForm.$valid) {
```

Comment:
An early return would make the code more readable - i.e. return if the form is not valid.